src/main/java/com/contrastsecurity/maven/plugin/ContrastInstallAgentMojo.java

```diff
@@ -336,52 +338,40 @@ String buildArgLine(String currentArgLine, String applicationName) {
-
-      final String organizationID = getOrganizationId();
-      try {
-        javaAgent = connection.getAgent(AgentType.JAVA, organizationID);
-      } catch (IOException e) {
-        throw new MojoExecutionException(
-            "\n\nCouldn't download the Java agent from Contrast. Please check that all your credentials are correct. If everything is correct, please contact Contrast Support. The error is:",
-            e);
-      } catch (UnauthorizedException e) {
-        throw new MojoExecutionException(
-            "\n\nWe contacted Contrast successfully but couldn't authorize with the credentials you provided. The error is:",
-            e);
-      }
-
-      // Save the jar to the 'target' directory
-      agentFile = new File(project.getBuild().getDirectory() + File.separator + AGENT_NAME);
-
-      try {
-        FileUtils.writeByteArrayToFile(agentFile, javaAgent);
-      } catch (IOException e) {
-        throw new MojoExecutionException("Unable to save the latest java agent.", e);
-      }
-
-      getLog().info("Saved the latest java agent to " + agentFile.getAbsolutePath());
-      contrastAgentLocation = agentFile.getAbsolutePath();
-
-    } else {
+  Path installJavaAgent(ContrastSDK connection) throws MojoExecutionException {
+    if (jarPath != null) {
       getLog().info("Using configured jar path " + jarPath);
-
-      agentFile = new File(jarPath);
-
-      if (!agentFile.exists()) {
+      final Path agent = Paths.get(jarPath);
+      if (!Files.exists(agent)) {
         throw new MojoExecutionException("Unable to load the local Java agent from " + jarPath);
       }
-
       getLog().info("Loaded the latest java agent from " + jarPath);
-      contrastAgentLocation = jarPath;
+      return agent;
     }
 
-    return agentFile;
+    getLog().info("No jar path was configured. Downloading the latest contrast.jar...");
+    final byte[] bytes;
+    final String organizationID = getOrganizationId();
+    try {
+      bytes = connection.getAgent(AgentType.JAVA, organizationID);
+    } catch (IOException e) {
+      throw new MojoExecutionException(
+          "\n\nCouldn't download the Java agent from Contrast. Please check that all your credentials are correct. If everything is correct, please contact Contrast Support. The error is:",
+          e);
+    } catch (UnauthorizedException e) {
+      throw new MojoExecutionException(
+          "\n\nWe contacted Contrast successfully but couldn't authorize with the credentials you provided. The error is:",
+          e);
+    }
+    // Save the jar to the 'target' directory
+    final Path agent = Paths.get(project.getBuild().getDirectory(), AGENT_NAME);
+    try {
+      Files.write(agent, bytes, StandardOpenOption.CREATE, StandardOpenOption.WRITE);
+    } catch (final IOException e) {
+      throw new MojoExecutionException("Unable to save the latest java agent.", e);
+    }
+    getLog().info("Saved the latest java agent to " + agent.toAbsolutePath());
```

Comment:
i think i just hate the java bean style, but i despise that this accessor grabs the injected logger. Might as well must make `logger` in `AbstractMojo` public or protected instead...

Reply:
Or not provide the logger via the base class at all